src/Internal/Test/E2E/Options.purs

```diff
@@ -0,0 +1,338 @@
+module Ctl.Internal.Test.E2E.Options where
+
+import Prelude
+
+import Control.Alt ((<|>))
+import Ctl.Internal.Test.E2E.Types
+  ( Browser
+  , ChromeUserDataDir
+  , CrxFilePath
+  , E2ETest
+  , ExtensionId
+  , SettingsArchive
+  , TmpDir
+  , WalletExt(LodeExt, FlintExt, GeroExt, NamiExt, EternlExt)
+  , WalletPassword
+  , mkE2ETest
+  , mkExtensionId
+  )
+import Data.Array (catMaybes)
+import Data.Array as Array
+import Data.Either (note)
+import Data.Foldable (fold)
+import Data.Generic.Rep (class Generic)
+import Data.List (List(Nil))
+import Data.Map (Map)
+import Data.Map as Map
+import Data.Maybe (Maybe(Nothing, Just))
+import Data.Show.Generic (genericShow)
+import Data.Tuple.Nested (type (/\), (/\))
+import Effect (Effect)
+import Node.Path (FilePath)
+import Options.Applicative
+  ( Parser
+  , ReadM
+  , command
+  , execParser
+  , fullDesc
+  , help
+  , helper
+  , info
+  , int
+  , long
+  , many
+  , metavar
+  , option
+  , progDesc
+  , showDefaultWith
+  , str
+  , subparser
+  , switch
+  , value
+  , (<**>)
+  )
+import Options.Applicative.Builder (eitherReader)
+import Record.Builder (build, merge)
+import Type.Row (type (+))
+
+-- | CLI options for E2E tests.
+type TestOptions = Record
+  (NoHeadless_ + Tests_ + TestTimeout_ + CommonOptions_ + ())
+
+type NoHeadless_ r = (noHeadless :: Boolean | r)
```

Comment:
Is the `r` parameter here to allow users to extend those options? since we don't provide a way to add new environment variables for them or command options, maybe this can be removed? I'm thinking in #734 here.

Reply:
No. It's just how `+` from type-level prelude works for record composition: https://pursuit.purescript.org/packages/purescript-typelevel-prelude/7.0.0/docs/Type.Row#t:type%20(+)